include/boost/thread/pthread/mutex.hpp

```diff
@@ -179,15 +194,26 @@ namespace boost
                 boost::throw_exception(thread_resource_error(res, "boost:: timed_mutex constructor failed in pthread_mutex_init"));
             }
 #ifndef BOOST_PTHREAD_HAS_TIMEDLOCK
+#ifdef BOOST_THREAD_HAS_CONDATTR_SET_CLOCK_MONOTONIC
```

Comment:
As this would appear several times, I believe it will be word defining an internal thread_detail::pthread_cond_init function that do this

Reply:
Good idea.